### PR TITLE
Fix test dependencies for subpackages and subpackage tests/examples enabless (#63, #268, #299, #510)

### PR DESCRIPTION
**Pull request by bartlettroscoe**
Addresses specifically the defect in #510 which resulted from work done as part of #63 and #299.  This also mostly addresses the scope of #268.

Once this is updated to Trilinos, it will fix https://github.com/trilinos/Trilinos/issues/10842.

See the specific git commit log messages for more detail.

### CODE REVIEW

**Review by rabartlett1972 — APPROVED** [review]
I am going to allow this to merge without a review because this is desperately needed to fix  https://github.com/trilinos/Trilinos/issues/10842 which will allow a bunch of PRs to pass Trilinos PR testing and be merged.

**Review by rabartlett1972 — APPROVED** [review]
Approving after the last commit to fix the tests.

**Comment by bartlettroscoe** [comment]
Hello @KyleFromKitware, can you please do a post-merge review of this PR?  Thanks!

File: tribits/core/package_arch/TribitsAdjustPackageEnables.cmake [hunk 1]
```diff
@@ -999,14 +999,59 @@ macro(tribits_apply_test_example_enables PACKAGE_NAME)
...
+      " ${PARENT_PACKAGE_NAME}_ENABLE_TESTS=${${PARENT_PACKAGE_NAME}_ENABLE_TESTS}")
+    set(${PARENT_PACKAGE_NAME}_ENABLE_EXAMPLES ${${PARENT_PACKAGE_NAME}_ENABLE_TESTS})
+  endif()
+  foreach(spkg IN LISTS ${PARENT_PACKAGE_NAME}_SUBPACKAGES)
+    set(fullSpkgName ${PARENT_PACKAGE_NAME}${spkg})
+    if (${PROJECT_NAME}_ENABLE_${fullSpkgName})
+      if (${PARENT_PACKAGE_NAME}_ENABLE_TESTS)
+        if ("${${fullSpkgName}_ENABLE_TESTS}" STREQUAL "")
+          message("-- " "Setting"
+	    " ${fullSpkgName}_ENABLE_TESTS=${${PARENT_PACKAGE_NAME}_ENABLE_TESTS}"
```
Comment (KyleFromKitware) [inline tribits/core/package_arch/TribitsAdjustPackageEnables.cmake:1022]: Not sure how picky you are about whitespace, but you've got some stray tabs here.

**Review by KyleFromKitware — COMMENTED** [review]
LGTM +1

File: tribits/core/package_arch/TribitsAdjustPackageEnables.cmake (hunk 1 above)
Comment (bartlettroscoe) [reply tribits/core/package_arch/TribitsAdjustPackageEnables.cmake:1022]: Initial tabs are a no-no.  Not sure how that got inserted (my emacs editor should be setup to indent with spaces only).   I will diffidently fix that.

Thanks!

Comment (bartlettroscoe) [reply tribits/core/package_arch/TribitsAdjustPackageEnables.cmake:1022]: This is addressed in new PR #515.  I found a bunch of other places where tabs somehow got inserted.  I wish there was a simple linter that could catch this.  (Actually, to look for initial tabs in `*.cmake` files, then that is easy.  I can do that with a simple bash scripts.)

**Pull request merged**